File: python/ABSTRACT_FACTORY_DECORATOR/ABSTRACT_FACTORY_DECORATOR/decorator/ReservacionDecoratorFactory.py
from ABSTRACT_FACTORY_DECORATOR.reservacion.ReservacionSencilla import ReservacionSencilla
from ABSTRACT_FACTORY_DECORATOR.reservacion.Reservacion import Reservacion
from ABSTRACT_FACTORY_DECORATOR.abstract_factory.ReservacionFactory import ReservacionFactory

class ReservacionDecoratorFactory(ReservacionFactory):
    
    reservacionSencilla  : ReservacionSencilla
    
    def __init__(self, reservacionSencilla) -> None:
        self.reservacionSencilla = reservacionSencilla
    
    def aumentarReservacion(self, datos):
        pass

File: php/ABSTRACT_FACTORY_DECORATOR/decorator/ReservacionDecoratorFactory.php
<?php

require_once './abstractFactory/ReservacionFactory.php';

class ReservacionDecoratorFactory implements ReservacionFactory{
    
    public $reservacionSencilla;
    
    public function __construct($reservacionSencilla) {
        $this->reservacionSencilla = $reservacionSencilla;
    }

    public function crearReservacion($datos) {
        
    }
    
    public function aumentarReservacion($datos) {
        
    }
    
    
    

}
